src/fprime/fbuild/target.py

```diff
@@ -39,7 +39,8 @@ def __init__(self, scope: TargetScope):
         """Set scope of this action"""
         self.scope = scope
 
-    def is_supported(self, builder: "Build", context: Path):
```

Comment:
My recollection when writing this file was that the subclasses overriding this need to access to self. Hence, the base method doesn't need a self-reference, but the overriding methods do.  I'd revert all three methods here.

Reply:
OK, actually, I suspected something like that and was hesitant on this base class. I will correct tomorrow, unless you want to have the pleasure of doing it ;)

Reply:
Done.